### PR TITLE
Running setup script outside of AWS

### PR DESCRIPTION
**Pull request by richardgill**
There's probably a bit too much stuff in here. You might want to not do all of it. I'm reasonably techy and wanted to do the course on my gaming rig to save a few $. 

- I tried this script with Ubuntu 16.10, it didn't quite work (nearly worked though!).
- Fixed home location in a couple of places (If this works on AWS this feels like a no brainer to me).
- My Theano wasn't picking up Cuda. Telling it where Cuda was fixed that (and probably won't hurt your AWS stuff).
- I'm running Ubuntu Desktop so I have to pair back my Graphics card to 75% (you might not want to do this one).

### CODE REVIEW

File: setup/install-gpu.sh [hunk 1]
```diff
@@ -40,8 +48,7 @@ sudo cp include/* /usr/local/cuda/include/
 
 jupyter notebook --generate-config
 jupass=`python -c "from notebook.auth import passwd; print(passwd())"`
-echo "c.NotebookApp.password = u'"$jupass"'" >> .jupyter/jupyter_notebook_config.py
+echo "c.NotebookApp.password = u'"$jupass"'" >> ~/.jupyter/jupyter_notebook_config.py
```
Comment (alexanderkjeldaas) [inline setup/install-gpu.sh]: better to use `$HOME` instead of `~` as then it works in `sh` as well.

File: setup/install-gpu.sh [hunk 2]
```diff
@@ -40,8 +48,7 @@ sudo cp include/* /usr/local/cuda/include/
 
 jupyter notebook --generate-config
 jupass=`python -c "from notebook.auth import passwd; print(passwd())"`
-echo "c.NotebookApp.password = u'"$jupass"'" >> .jupyter/jupyter_notebook_config.py
+echo "c.NotebookApp.password = u'"$jupass"'" >> ~/.jupyter/jupyter_notebook_config.py
 echo "c.NotebookApp.ip = '*'
-c.NotebookApp.open_browser = False" >> .jupyter/jupyter_notebook_config.py
+c.NotebookApp.open_browser = False" >> ~/.jupyter/jupyter_notebook_config.py
```
Comment (alexanderkjeldaas) [inline setup/install-gpu.sh]: and here.

File: setup/install-gpu.sh [hunk 3]
```diff
@@ -13,15 +15,21 @@ mkdir downloads
...
 conda upgrade -y --all
 
 pip install theano
 echo "[global]
 device = gpu
-floatX = float32" > ~/.theanorc
+floatX = float32
+
+[lib]
+cnmem=.75
```
Comment (racheltho) [inline setup/install-gpu.sh]: We don't want to use cnmem by default

File: setup/install-gpu.sh [hunk 4]
```diff
@@ -13,15 +15,21 @@ mkdir downloads
...
+
+[cuda]
+root = /usr/local/cuda" > ~/.theanorc1
```
Comment (racheltho) [inline setup/install-gpu.sh]: I think the 1 at the EOL is a typo

**Comment by richardgill** [comment]
Made those changes, thanks.

**Pull request merged**